### PR TITLE
Card script for Call to the Void and changes to ChooseCardEffect to enable card mechanisms.

### PR DESCRIPTION
**Pull request by xaviermd**
- Created CardScript for call to the void.
- Added "ControlAndNot" parameter to ChooseCardEffect as suggested by @northmoc on discord (https://discord.com/channels/267367946135928833/994588726254633090/994610830815539370)
- Added and implemented "SecretlyChoose" in ChooseCardEffect.
- Added RevealTitle to ChooseCardEffect.

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/call_to_the_void.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Name:Call to the Void
+ManaCost:4 B
+Types:Sorcery
+A:SP$ ChooseCard | Defined$ Player | Choices$ Creature | SecretlyChoose$ True | Amount$ 1 | ControlAndNot$ True | ChoiceTitle$ Secretly choose a creature | Reveal$ True | RevealTitle$ OVERRIDE Chosen creatures. They will be destroyed. | SubAbility$ DBDestroyChosen | RememberChosen$ True | Mandatory$ True
+SVar:DBDestroyChosen:DB$ DestroyAll | ValidCards$ Creature.IsRemembered | SubAbility$ DBCleanUp
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/call_to_the_void.txt:5]: You might not need IsRemembered and just use ChosenCard instead

File: forge-game/src/main/java/forge/game/ability/effects/ChooseCardEffect.java [hunk 2]
```diff
@@ -83,7 +83,7 @@ public void resolve(SpellAbility sa) {
             combined.addAll(choices);
             choices = combined;
         }
-
+
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/ability/effects/ChooseCardEffect.java:86]: Nice work, just some spaces here that should be removed

File: forge-gui/res/cardsfolder/upcoming/call_to_the_void.txt [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+SVar:DBCleanUp:DB$ Cleanup | ClearRemembered$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/call_to_the_void.txt:6]: Missing Oracle:

File: forge-gui/res/cardsfolder/upcoming/call_to_the_void.txt (hunk 1 above)
Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/call_to_the_void.txt:5]: I agree!

File: forge-gui/res/cardsfolder/upcoming/call_to_the_void.txt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Name:Call to the Void
+ManaCost:4 B
+Types:Sorcery
+A:SP$ ChooseCard | Defined$ Player | Choices$ Creature | SecretlyChoose$ True | Amount$ 1 | ControlAndNot$ True | ChoiceTitle$ Secretly choose a creature | Reveal$ True | RevealTitle$ OVERRIDE Chosen creatures. They will be destroyed. | SubAbility$ DBDestroyChosen | Mandatory$ True
+SVar:DBDestroyChosen:DB$ DestroyAll | ValidCards$ Creature.ChosenCard | SubAbility$ DBCleanUp
+SVar:DBCleanUp:DB$ Cleanup | ClearRemembered$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/call_to_the_void.txt:6]: Now you no longer need the Cleanup :)

Comment (xaviermd) [reply forge-gui/res/cardsfolder/upcoming/call_to_the_void.txt]: I had actually reviewed the script a few times to try to make sure I wasn't forgetting anything this time...😅 Oh well! Thank you for the feedback!

**Review by tool4ever — APPROVED** [review]

**Comment by xaviermd** [comment]
closes #818

**Pull request merged**